extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -148,7 +148,20 @@ private Uni<SecurityIdentity> createSecurityIdentityWithOidcServer(RoutingContex
                 tokenUni = verifySelfSignedTokenUni(resolvedContext, request.getToken().getToken());
             }
         } else {
-            tokenUni = verifyTokenUni(resolvedContext, request.getToken().getToken());
+            // verify opaque access token with UserInfo if enabled
+            if (resolvedContext.oidcConfig != null && resolvedContext.oidcConfig.verifyOpaqueAccessTokenWithUserInfo
```

Comment:
I think we need to push this check down to `verifyTokenUni` as we don't now here if the introspection path is available or not - could have been auto-discovered, pass UserInf there as well. No need to check for the oidc config being null - would be a bug if it were. The opaque token check is done below as well, so inside that branch you can check if the opaque access token can be verified only if introspection path is null and userinfo is not null